### PR TITLE
New API endpoints for managed keys

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- info about the managed keys should be exported while in multikey mode
  
## Proposed changes
- `node/managed-keys/count` for the total number of managed keys
- `node/managed-keys/eligible/:epoch` for the eligible keys managed, in a specific epoch, from the same shard as the node
- `node/managed-keys/waiting/:epoch` for the waiting keys managed, in a specific epoch, from the same shard as the node

## Testing procedure
- will be tested with feat branch

### Response examples:
image
image

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: api/groups/nodeGroup.go [hunk 1]
```diff
@@ -338,6 +359,67 @@ func (ng *nodeGroup) connectedPeersRatings(c *gin.Context) {
...
+
+// managedKeysEligible returns the node's eligible managed keys
+func (ng *nodeGroup) managedKeysEligible(c *gin.Context) {
+	epoch, err := getQueryParamEpoch(c)
+	if err != nil {
+		shared.RespondWithValidationError(c, errors.ErrValidation, errors.ErrBadUrlParams)
+		return
+	}
+
+	keys, err := ng.getFacade().GetEligibleManagedKeys(epoch)
```
Comment (iulianpascalau) [inline api/groups/nodeGroup.go:383]: this works for older epochs but won't work for future epochs. Do you think it is relevant to keep the epoch parameter? I think the use case will be to check the current epoch status.

File: api/groups/nodeGroup.go [hunk 2]
```diff
@@ -338,6 +359,67 @@ func (ng *nodeGroup) connectedPeersRatings(c *gin.Context) {
...
+	keys, err := ng.getFacade().GetWaitingManagedKeys(epoch)
+	if err != nil {
+		shared.RespondWithInternalError(c, errors.ErrGetWaitingManagedKeys, err)
+		return
+	}
+
+	c.JSON(
+		http.StatusOK,
+		shared.GenericAPIResponse{
+			Data:  gin.H{"keys": keys},
```
Comment (BeniaminDrasovean) [inline api/groups/nodeGroup.go:416]: waitingKeys?

File: testscommon/managedPeersMonitorStub.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+// GetWaitingManagedKeys -
+func (stub *ManagedPeersMonitorStub) GetWaitingManagedKeys(epoch uint32) ([][]byte, error) {
+	if stub.GetWaitingManagedKeysCalled != nil {
+		return stub.GetWaitingManagedKeysCalled(epoch)
+	}
+	return make([][]byte, 0), nil
+}
+
```
Comment (BeniaminDrasovean) [inline testscommon/managedPeersMonitorStub.go:33]: Missing comment.

File: api/groups/nodeGroup.go (hunk 1 above)
Comment (sstanculeanu) [reply api/groups/nodeGroup.go:383]: right, would make more sense.. updated

File: api/groups/nodeGroup.go [hunk 4]
```diff
@@ -338,6 +359,67 @@ func (ng *nodeGroup) connectedPeersRatings(c *gin.Context) {
...
+	keys, err := ng.getFacade().GetEligibleManagedKeys(epoch)
+	if err != nil {
+		shared.RespondWithInternalError(c, errors.ErrGetEligibleManagedKeys, err)
+		return
+	}
+
+	c.JSON(
+		http.StatusOK,
+		shared.GenericAPIResponse{
+			Data:  gin.H{"keys": keys},
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply api/groups/nodeGroup.go:392]: updated

File: api/groups/nodeGroup.go (hunk 2 above)
Comment (sstanculeanu) [reply api/groups/nodeGroup.go:416]: updated

File: testscommon/managedPeersMonitorStub.go (hunk 3 above)
Comment (sstanculeanu) [reply testscommon/managedPeersMonitorStub.go:33]: added

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**